Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,297 @@
+            SetTestStateSkipped
+            exit 0
+            ;;
+    esac
+}
+
+function verify_expected_file() {
+    if [ ! -f "$1" ]; then
+        fail_test "${1} doesn't exist."
+    fi
+}
+
+function verify_da_pid() {
+    if [ ! -f $da_pid_file ]; then
+        fail_test "$da_pid_file does not exist."
+    fi
+
+    if ! x=$(sed -n "/^[1-9][0-9]*$/p;q" $da_pid_file); then
+        fail_test "PID not found in DA_PID"
+    fi
+    
+    if strings /proc/$x/cmdline | fgrep -q  -e "microsoft-dependency-agent-manager"; then
+        LogMsg "PID matched"
+    else
+        fail_test "PID not matched"
+    fi
+}
+
+function check_prereqs() {
+    LogMsg "Checking for preconditions"
+
+    if [ $EUID -ne 0 ]; then
+        LogErr "This script must be run as root: $EUID"
+        SetTestStateAborted
+        exit 0
+    fi
+
+    verify_distro
+
+    if check_da_directories_exist; then
+        SetTestStateAborted
+        exit 0
+    fi
+
+    LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+    LogMsg "Download Dependency Agent Linux installer"
+
+    if ! curl -L https://aka.ms/dependencyagentlinux -o "$da_installer"; then
+        fail_test Download failed
+    fi
+
+    chmod +x "$da_installer"
+}
+
+function verify_install_da() {
+    LogMsg "Starting Install tests"
+
+    "$da_installer" -vme
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Install failed with exit code ${ret}"
+    fi
+    LogMsg "Dependency Agent installed successfully"
+
+    verify_expected_file "$uninstaller"
+    verify_expected_file "/etc/init.d/microsoft-dependency-agent"
+    verify_expected_file "$da_log_dir/install.log"
+
+    bin_version=$("$da_installer" --version | awk "{print $5}")
+    install_log_version=$(sed -n "s/^Dependency Agent version\.revision: //p" $da_log_dir/install.log)
+    if [ "$bin_version" -ne "$install_log_version" ]; then
+        fail_test "Version mismatch between bin version($bin_version) and install log version($install_log_version)"
+    fi
+    LogMsg "Version matches between bin version and install log version"
+
+    LogMsg "Install tests completed"
+}
+
+function verify_uninstall_da() {
+    LogMsg "Starting uninstall tests"
+
+    $uninstaller
+
+    ret=$?
+    if [ $ret -ne 0 ]; then
+        fail_test "Uninstall failed with exit code ${ret}"
+    else
+        LogMsg "Uninstalled with exit code 0"
+    fi
+
+    if check_da_directories_exist; then
+        fail_test "Directory exists"
+    fi
+
+    LogMsg "Uninstall tests completed"
+}
+
+function generate_network_activity() {
+    LogMsg "Generate network activity"
+    watch -n 5 curl https://microsoft.com &>/dev/null &
+    network_activity_generator_pid=$!
+}
+
+function enable_disable_da(){
+    LogMsg "Starting Enable/Disable DA tests"
+
+    verify_install_da
+
+    # Wait for 30s for DA to start running and check for service.log
+    sleep 30
+    verify_expected_file "$da_log_dir/service.log"
```

Comment:
OOPS! This service log assertion is always going to fail. When the installer is run with the `-vme` flag, as specified for Test Case 1, it only does the installation. Test Case 2 specified the `-s` flag (silent install) that does the install and starts/enables the agent. Without the agent starting there will not be a service.log.

Reply:
Done. Parameterized `verify_install_da` function to handle this